src/transformers/generation_utils.py

```diff
@@ -392,6 +392,8 @@ class GenerationMixin:
           `do_sample=False`.
         - *multinomial sampling* by calling [`~generation_utils.GenerationMixin.sample`] if `num_beams=1` and
           `do_sample=True`.
+        - *typical decoding* by calling [`~generation_utils.GenerationMixin.sample`] if `typical_p` between 0 and 1,
+          and `do_sample=True`.
```

Comment:
This part of the docstring is reserved to separate functions (such as `generation_utils.GenerationMixin.sample`). Since typical decoding is done inside `sample`, it can't be listed here :) 

A good place to augment the documentation regarding typical decoding would be in the [corresponding logit processor](https://github.com/huggingface/transformers/blob/2df602870b9a7403109b470322926b48e366e3d1/src/transformers/generation_logits_process.py#L238), and I would 100% support it!